### ISSUE TITLE
Bug: Mocking annotation causes MockkException

### ISSUE THREAD

**Issue by ThanksForAllTheFish** [mockk/mockk#226]
The following code cause a `MockkException`:

```
import io.kotlintest.specs.ExpectSpec
import io.mockk.mockk

interface AnInterface
annotation class BlaBla

class MustExistValidatorTest : ExpectSpec() {
    val blaBla = mockk<BlaBla>()
}
```

Stacktrace:

```
	at io.mockk.proxy.jvm.transformation.SubclassInstrumentation.access$doInterceptedSubclassing(SubclassInstrumentation.kt:17)
java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:488)
	at kotlin.reflect.jvm.internal.calls.CallerImpl$Constructor.call(CallerImpl.kt:41)
	at kotlin.reflect.jvm.internal.KCallableImpl.call(KCallableImpl.kt:106)
	at kotlin.reflect.jvm.internal.KCallableImpl.callDefaultMethod$kotlin_reflect_api(KCallableImpl.kt:152)
	at kotlin.reflect.jvm.internal.KCallableImpl.callBy(KCallableImpl.kt:110)
	at kotlin.reflect.full.KClasses.createInstance(KClasses.kt:283)
	at io.kotlintest.runner.jvm.JvmKt.instantiateSpec(jvm.kt:12)
	at io.kotlintest.runner.jvm.TestEngine.createSpec(TestEngine.kt:100)
	at io.kotlintest.runner.jvm.TestEngine.access$createSpec(TestEngine.kt:15)
	at io.kotlintest.runner.jvm.TestEngine$submitSpec$1.run(TestEngine.kt:78)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:514)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1135)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:844)
Caused by: io.mockk.MockKException: Can't instantiate proxy for class ch.olmero.masterplan.validation.BlaBla
	at io.mockk.impl.instantiation.JvmMockFactory.newProxy(JvmMockFactory.kt:64)
	at io.mockk.impl.instantiation.AbstractMockFactory.newProxy$default(AbstractMockFactory.kt:29)
	at io.mockk.impl.instantiation.AbstractMockFactory.mockk(AbstractMockFactory.kt:58)
	at ch.olmero.masterplan.validation.MustExistValidatorTest.<init>(MustExistValidatorTest.kt:28)
	... 18 more
Caused by: io.mockk.proxy.MockKAgentException: Failed to subclass interface ch.olmero.masterplan.validation.BlaBla
	at io.mockk.proxy.jvm.ProxyMaker.proxy(ProxyMaker.kt:38)
	at io.mockk.impl.instantiation.JvmMockFactory.newProxy(JvmMockFactory.kt:34)
	... 21 more
Caused by: java.lang.IllegalArgumentException: Could not create type
	at net.bytebuddy.TypeCache.findOrInsert(TypeCache.java:154)
	at net.bytebuddy.TypeCache.findOrInsert(TypeCache.java:174)
	at io.mockk.proxy.jvm.transformation.SubclassInstrumentation.subclass(SubclassInstrumentation.kt:46)
	at io.mockk.proxy.jvm.ProxyMaker.subclass(ProxyMaker.kt:113)
	at io.mockk.proxy.jvm.ProxyMaker.proxy(ProxyMaker.kt:35)
	... 22 more
Caused by: java.lang.NoClassDefFoundError: java/lang/annotation/Annotation (wrong name: java/lang/WeakPairMap$Pair$Weak$1)
	at java.base/java.lang.WeakPairMap$Pair$Weak.<init>(WeakPairMap.java:264)
	at java.base/java.lang.WeakPairMap$Pair.weak(WeakPairMap.java:201)
	at java.base/java.lang.WeakPairMap.putIfAbsent(WeakPairMap.java:123)
	at java.base/java.lang.Module.implAddReads(Module.java:415)
	at java.base/java.lang.Module.implAddReads(Module.java:376)
	at java.base/java.lang.System$2.addReads(System.java:2157)
	at java.base/jdk.internal.module.Modules.addReads(Modules.java:85)
	at java.base/java.lang.reflect.Proxy$ProxyBuilder.lambda$getDynamicModule$4(Proxy.java:884)
	at java.base/jdk.internal.loader.AbstractClassLoaderValue$Memoizer.get(AbstractClassLoaderValue.java:329)
	at java.base/jdk.internal.loader.AbstractClassLoaderValue.computeIfAbsent(AbstractClassLoaderValue.java:205)
	at java.base/java.lang.reflect.Proxy$ProxyBuilder.getDynamicModule(Proxy.java:875)
	at java.base/java.lang.reflect.Proxy$ProxyBuilder.mapToModule(Proxy.java:814)
	at java.base/java.lang.reflect.Proxy$ProxyBuilder.<init>(Proxy.java:631)
	at java.base/java.lang.reflect.Proxy$ProxyBuilder.<init>(Proxy.java:636)
	at java.base/java.lang.reflect.Proxy.lambda$getProxyConstructor$0(Proxy.java:415)
	at java.base/jdk.internal.loader.AbstractClassLoaderValue$Memoizer.get(AbstractClassLoaderValue.java:329)
	at java.base/jdk.internal.loader.AbstractClassLoaderValue.computeIfAbsent(AbstractClassLoaderValue.java:205)
	at java.base/java.lang.reflect.Proxy.getProxyConstructor(Proxy.java:413)
	at java.base/java.lang.reflect.Proxy.newProxyInstance(Proxy.java:999)
	at java.base/sun.reflect.annotation.AnnotationParser$1.run(AnnotationParser.java:305)
	at java.base/sun.reflect.annotation.AnnotationParser$1.run(AnnotationParser.java:303)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/sun.reflect.annotation.AnnotationParser.annotationForMap(AnnotationParser.java:303)
	at java.base/sun.reflect.annotation.AnnotationParser.parseAnnotation2(AnnotationParser.java:293)
	at java.base/sun.reflect.annotation.AnnotationParser.parseAnnotations2(AnnotationParser.java:120)
	at java.base/sun.reflect.annotation.AnnotationParser.parseAnnotations(AnnotationParser.java:72)
	at java.base/java.lang.reflect.Executable.declaredAnnotations(Executable.java:605)
	at java.base/java.lang.reflect.Executable.declaredAnnotations(Executable.java:603)
	at java.base/java.lang.reflect.Executable.getDeclaredAnnotations(Executable.java:591)
	at java.base/java.lang.reflect.Constructor.getDeclaredAnnotations(Constructor.java:579)

	at net.bytebuddy.description.method.MethodDescription$ForLoadedConstructor.getDeclaredAnnotations(MethodDescription.java:1081)
	at io.mockk.proxy.jvm.transformation.SubclassInstrumentation$subclass$1.call(SubclassInstrumentation.kt:49)
	at net.bytebuddy.description.method.MethodDescription$TypeSubstituting.getDeclaredAnnotations(MethodDescription.java:1644)

	at io.mockk.proxy.jvm.transformation.SubclassInstrumentation$subclass$1.call(SubclassInstrumentation.kt:17)
	at net.bytebuddy.TypeCache.findOrInsert(TypeCache.java:152)

	... 26 common frames omitted	at net.bytebuddy.description.method.MethodDescription$AbstractBase.asToken(MethodDescription.java:819)
	at net.bytebuddy.description.method.MethodDescription$AbstractBase.asToken(MethodDescription.java:376)
	at net.bytebuddy.description.method.MethodList$AbstractBase.asTokenList(MethodList.java:70)
	at net.bytebuddy.dynamic.scaffold.subclass.ConstructorStrategy$Default$5.doExtractConstructors(ConstructorStrategy.java:175)
	at net.bytebuddy.dynamic.scaffold.subclass.ConstructorStrategy$Default.extractConstructors(ConstructorStrategy.java:196)
	at net.bytebuddy.dynamic.scaffold.subclass.SubclassDynamicTypeBuilder.applyConstructorStrategy(SubclassDynamicTypeBuilder.java:223)
	at net.bytebuddy.dynamic.scaffold.subclass.SubclassDynamicTypeBuilder.make(SubclassDynamicTypeBuilder.java:198)
	at net.bytebuddy.dynamic.scaffold.subclass.SubclassDynamicTypeBuilder.make(SubclassDynamicTypeBuilder.java:189)
	at net.bytebuddy.dynamic.DynamicType$Builder$AbstractBase.make(DynamicType.java:3394)
	at net.bytebuddy.dynamic.DynamicType$Builder$AbstractBase$Delegator.make(DynamicType.java:3583)
	at io.mockk.proxy.jvm.transformation.SubclassInstrumentation.doInterceptedSubclassing(SubclassInstrumentation.kt:79)
	at io.mockk.proxy.jvm.transformation.SubclassInstrumentation.access$doInterceptedSubclassing(SubclassInstrumentation.kt:17)
	at io.mockk.proxy.jvm.transformation.SubclassInstrumentation$subclass$1.call(SubclassInstrumentation.kt:49)
	at io.mockk.proxy.jvm.transformation.SubclassInstrumentation$subclass$1.call(SubclassInstrumentation.kt:17)
	at net.bytebuddy.TypeCache.findOrInsert(TypeCache.java:152)
	... 26 more
```

On the other hand, the following code runs smoothly:

```
import io.kotlintest.specs.ExpectSpec
import io.mockk.mockk

interface AnInterface
annotation class MyAnn

class MustExistValidatorTest : ExpectSpec() {
    val anInterface = mockk<AnInterface>()
    val blaBla = mockk<MyAnn>()
}
```

Tested with `mockk` `1.8.13` and `1.9`

**Comment by oleksiyp** [comment]
Managed to reproduce on Java 9

**Comment by oleksiyp** [comment]
It is race condition for initialization in WeekMap. Have very similar case here #168

**Comment by oleksiyp** [comment]
I added a few more classes to preload. Hopefully, this will help to solve it.
https://github.com/mockk/mockk/commit/5ed474fd2c636bfd324e8057606514325c5964bb

**Comment by oleksiyp** [comment]
Released [1.9.1](https://github.com/mockk/mockk/releases/tag/1.9.1). Please try it out and close the ticket if everything is fine. The fix itself is fixing the well known old issue related to a race condition, at least this one was seen in stacktrace. Maybe something else is needed to be fixed, just let me know

**Comment by ThanksForAllTheFish** [comment]
Sorry, I missed your message a month ago. `1.9.1` seems to fix the issue, I was able to replace the workaround and my test is still ok. Thanks for the fix!
